velox/functions/sparksql/String.h

```diff
@@ -1181,4 +1181,55 @@ struct FindInSetFunction {
   }
 };
 
+template <typename T>
+struct SoundexFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  static constexpr bool is_default_ascii_behavior = true;
+  static constexpr int32_t reuse_strings_from_arg = 0;
+
+  void call(out_type<Varchar>& result, const arg_type<Varchar>& input) {
+    auto inputData = reinterpret_cast<const uint8_t*>(input.data());
+    size_t inputSize = input.size();
+    if (inputSize == 0) {
```

Comment:
Move this check before `auto inputData = reinterpret_cast<const uint8_t*>(input.data());`

Reply:
Upated, thanks.